### PR TITLE
PROD-319/rename-campaigns-to-stacks 

### PR DESCRIPTION
**Pull request by danijel-radomiljac**

### CODE REVIEW

**Review by Yihwan — COMMENTED** [review]
just want to triple check that we have set up redirects and tested those from campaigns -> stacks to avoid any broken link

**Review by marvinmarnold — CHANGES_REQUESTED** [review]
- Can you update this to use a path defined here: https://github.com/gator-labs/chomp/blob/main/lib/urls.ts Generally, lets stop hardcoding paths in the app moving forward.
- It looks like /campaigns has been renamed /stacks. But like yihwan said, it would be good to have redirects from /campaigns -> /stacks as well. That way any old links from Breakpoint will keep working.

**Comment by danijel-radomiljac** [comment]
> * Can you update this to use a path defined here: [`main`/lib/urls.ts](https://github.com/gator-labs/chomp/blob/main/lib/urls.ts?rgh-link-date=2024-09-24T18%3A13%3A29Z) Generally, lets stop hardcoding paths in the app moving forward.
> * It looks like /campaigns has been renamed /stacks. But like yihwan said, it would be good to have redirects from /campaigns -> /stacks as well. That way any old links from Breakpoint will keep working.

Fixed ✅

**Review by marvinmarnold — CHANGES_REQUESTED** [review]
@danijel-radomiljac I think we need to keep the /campaigns routes that were publicly visible and add hard redirects to /stacks. Let me know if I'm missing some other way that people who see an old QR code to app.chomp.games/campaigns will manage to get redirected to /stacks.

This is in addition to the work in this PR which is good. I'm just saying add a step for redirects.

File: next.config.js [hunk 1]
```diff
@@ -24,6 +24,11 @@ const nextConfig = {
         destination: "/application",
         permanent: true,
       },
+      {
+        source: "/application/campaigns",
```
Comment (marvinmarnold) [inline next.config.js:28]: Shouldn't this be `/campaigns` -> `/stacks`?

File: app/components/CampaignsHeader/CampaignsHeader.tsx [hunk 2]
```diff
@@ -1,21 +1,22 @@
...
 import { useRouter } from "next/navigation";
 import { ArrowLeftIcon } from "../Icons/ArrowLeftIcon";
 
 interface CampaignsHeaderProps {
-  heading?: string; 
+  heading?: string;
   backAction: "back" | "campaigns";
 }
 
 const CampaignsHeader: React.FC<CampaignsHeaderProps> = ({
```
Comment (marvinmarnold) [inline app/components/CampaignsHeader/CampaignsHeader.tsx:12]: Might be worth going through and renaming the classes to, in order to reduce future tech debt. Eg rename to `StacksHeader` and `StacksHeaderProps`. Also all other campaigns classes in other files.

File: app/components/CampaignLeaderboard/CampaignLeaderboard.tsx [hunk 3]
```diff
@@ -8,7 +8,7 @@ const CampaignLeaderboard = async () => {
 
   return (
     <div className="flex flex-col gap-2">
-      <p>Campaign Leaderboard</p>
+      <p>Stack Leaderboard</p>
```
Comment (marvinmarnold) [inline app/components/CampaignLeaderboard/CampaignLeaderboard.tsx:11]: @Yihwan should this be `Stacks` (plural)?

**Review by marvinmarnold — CHANGES_REQUESTED** [review]

**Comment by marvinmarnold** [comment]
@danijel-radomiljac conflicts FYI

**Comment by marvinmarnold** [comment]
Test is failing because migration needs to run

**Pull request merged**